src/NServiceBus.Core/Routing/UnicastPublishRouter.cs

```diff
@@ -48,7 +50,11 @@ HashSet<string> SelectDestinationsForEachEndpoint(IDistributionPolicy distributi
                 }
                 else
                 {
-                    var subscriber = distributionPolicy.GetDistributionStrategy(group.First().Endpoint, DistributionStrategyScope.Publish).SelectReceiver(group.Select(s => s.TransportAddress).ToArray());
+                    var endpointName = @group.First().Endpoint;
+
+                    var instances = endpointInstances.FindInstances(endpointName).Select(e => transportAddressTranslation(e)).ToArray();
```

Comment:
So these are the cases:
 * If there is a V5 endpoint subscribed (without Endpoint value), it will get the event
 * If there is a number of V6 endpoint instances subscribed with same Endpoint name but that endpoint has no instances configured in the mapping file, the message will be send to the default instance (assuming queue on local machine)
 * if there is configuration, it will work using the round robin thing

The thing we are trading off here is configuration-less sender-side distribution for pubishes. The thing we are gaining is consistency between sends and publishes in the way they use the routing file e.g. you remove an instance from the routing file, it will not get events. In the current version it would still get the events as long as is not removed from subscription store.